monashspa/PHS3302/d0_lifetime/d0_lifetime.py

```diff
@@ -0,0 +1,410 @@
+# $$
+# 
+# Values of the pion and kaon masses have been measured elswhere:
+# - $m(\pi) = 139.6\,{\rm MeV}/c^{2}$
+# - $m(K) = 493.7\,{\rm MeV}/c^{2}$ 
+# 
+# **Task 1:** *[2 marks]* Using the example below, create a new variable called `K_E` and `Pi_E` in which you calculate the pion and kaon energies from the 3-momenta and their known masses.
+
+
+
+# Pi_E = 
+# K_E = 
+
+
+# Now we have the pion and kaon energies we have the complete 4-momentum for the kaon and pion
+# 
+# $$ 
+# \mathbf{p}_{\pi} = \left[E_\pi,p_{x}(\pi),p_{y}(\pi),p_{z}(\pi)\right]
+# $$
+# $$
+# \mathbf{p}_{K} = \left[E(K),p_{x}(K),p_{y}(K),p_{z}(K)\right]
+# $$
+# we can use these to calculate the mass of the $D^0$ meson candidate. 
+# 
+# 
+# The mass can be calculated from the sum of the momentum 4-vectors for the kaon and pion 
+# $$
+# \mathbf{p}_{D^{0}} = \mathbf{p}_{K} + \mathbf{p}_{\pi},
+# $$
+# 
+# In the rest frame of the $D^0$ meson we will have $\mathbf{p}_{D^0} = \left[m_{D^0},0,0,0\right]$
+# $$
+# m_{D^{0}}^2 = \mathbf{p}_{D^{0}}.\mathbf{p}_{D^{0}} = (\mathbf{p}_{K} + \mathbf{p}_{\pi})^2.
+# $$
+# **Task 2:** *[2 marks]* Calculate the $D^0$ mass using the all of the components of the kaon and pion 4-momentum. 
+
+
+
+## D0_M = 
+
+
+# A loose selection has already been applied to the sample we are analysing. This means that the signal contains both real $D^0\to K \pi$ decays as well as backgrounds from combinations of a kaon and pion that didn't originate from a $D^0$ meson. 
+# These two different contributions will be easy to see if you've calculated the $D^0$ mass correctly. Hint: the mass of the $D^0$ meson is $1864.84\pm0.05\,{\rm MeV}/c^2$. 
+# 
+# **Task 3:** *[2 marks]* Create a histogram of the $D^0$ mass.
+
+
+
+## Plot D0_M histogram
+
+
+# ### Part 2b: $D^0$ lifetime
+# To calculate the lifetime we first need to determine the distance between the proton-proton interaction vertex ($D^0$ production vertex, or primary vertex) and the $D^0$ decay vertex (or secondary vertex). The 3-vector between these points is given by:
+# $$
+# \Delta\vec{x} = \vec{x}_{\rm decay} - \vec{x}_{\rm production}.
+# $$
+# Note, both the production and decay positions are given in units of $\rm mm$. 
+# 
+# **Task 4:** *[4 marks]* Calculate and plot a histgram of the absolute distance $|\Delta\vec{x}|$ between the primary vertex (proton-proton collision) and secondary vertex ($D^0$ decay).
+
+
+
+# D0_L = 
+
+
+
+
+## Plot D0_L histogram
+
+
+# In the lab frame, a particle travelling at velocity $\vec{\mathbf{v}}$ for a time $t$ will travel a distance $\Delta \vec{\mathbf{x}}$:
+# $$
+# \Delta\vec{\mathbf{x}} = \vec{\mathbf{v}}t
+# $$
+# 
+# We must account for special relativity when trying to determine the lifetime of the $D^0$ meson. We can rewrite this equation using relations between the momentum and velocity, and between lab frame time and proper time:
+# 
+# $$
+# \vec{\mathbf{p}} = \gamma m \vec{\mathbf{v}}\\
+# t = \gamma \tau.
+# $$
+# 
```

Comment:
If you just remind them of this, and encourage them to carry the 3-vectors of p and v forward, they should be able to get the expression themselves.

Reply:
Good idea, I've created a new task asking them to do the last two steps themselves.